catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+        LOGGER.trace("Ignoring this part, since 100KM is only valid after a ZONE");
+        break;
+    }
+  }
+
+  private static void handleNumericWithPreviousType(
+      String mgrsPart,
+      MgrsPartType previousType,
+      StringBuilder mgrsBuilder,
+      List<String> mgrsStrings) {
+    LOGGER.trace("Found a NUMERIC [{}] which occurs after a {}", mgrsPart, previousType);
+    switch (previousType) {
+      case ZONE:
+      case HUNDRED_KM:
+      case NUMERIC:
+        String processed = processNumericType(mgrsPart);
+        LOGGER.trace("NUMERIC [{}] processed = [{}]", mgrsPart, processed);
+        LOGGER.trace("Storing final result coordinate");
+        mgrsStrings.add(mgrsBuilder.toString() + processed);
+        break;
+      case NONE:
+        LOGGER.trace(
+            "Ignoring this part, since NUMERIC is only valid in a ZONE but no ZONE was provided");
+        break;
+    }
+  }
+
+  /**
+   * Given an MGRS {@link MgrsPartType#NUMERIC} part format the string to ensure an accurate
+   * conversion, using whitespace as a hint as to the user's intentions.
+   *
+   * <p>The usng4j library demands a minimum of 6 symbols if the easting / northing data is not to
+   * be ignored. That is, the easting component must be fully specified for the conversion to work.
+   * To accomodate this, zeros are prepended onto the easting to whatever appropriate degree.
+   *
+   * <p>For example, an input of <span>4Q KJ 1 2</span> will become <span>4QKJ000012</span> so the
+   * easting / northing is not ignored.
+   *
+   * <p>Refer to the unit tests for more comprehensive examples.
+   *
+   * <p>TODO DDF-4243 - This should be simplified if usng4j is improved
+   *
+   * @see #assemble(String, String)
+   * @param mgrsPart an MGRS substring that contains BOTH the easting and northing.
+   * @return a modified easting/northing string with zeros backfilled to ensure accurate conversion.
+   */
+  private static String processNumericType(String mgrsPart) {
+    String[] numericParts = mgrsPart.split("[ ]+");
+    if (numericParts.length >= 3 || numericParts.length == 0) {
+      throw new IllegalArgumentException(
+          "The mgrs part provided is not a valid NUMERIC type: " + mgrsPart);
+    }
+    if (numericParts.length == 2) {
+      return assemble(numericParts[0], numericParts[1]);
+    }
+    String eastingAndNorthing = numericParts[0];
+    if (eastingAndNorthing.length() % 2 > 0) {
+      eastingAndNorthing = eastingAndNorthing.substring(0, eastingAndNorthing.length() - 1);
+    }
+    int halfway = eastingAndNorthing.length() / 2;
+    String easting = eastingAndNorthing.substring(0, halfway);
+    String northing = eastingAndNorthing.substring(halfway, eastingAndNorthing.length());
+    return assemble(easting, northing);
+  }
+
+  private static String assemble(String easting, String northing) {
+    StringBuilder numericBuilder = new StringBuilder();
+    Runnable appendZero = () -> numericBuilder.append("0");
```

Comment:
❓ Why can't we do `new StringBuilder().append("0");`

Reply:
Because the number of zeros we want to append is variable. This runnable represents the operation as a whole, and is passed to the `repeat(...)` function to be done `numberOfZerosNeeded` times. We're backfilling zeros for the easting and the northing.

Reply:
Can't we use `String.format` with padded?  Runnable seems like overkill.

Reply:
https://github.com/codice/ddf/pull/3901#discussion_r227447004

Reply:
Note per the javadoc the padding of zeros is done prior to feeding the input into a conversion library. This library has pretty strict enforcement on the format it expects. This was done to ensure the result is always accurate per the string we are interpreting from user input. 

I can get rid of the `Runnable` so this is easier to consume.

Reply:
@bdeining I don't think a `format` string will make a difference. We have to iterate one way or another because the number of required zeros is not known (cannot be a constant).

Reply:
I believe a lot of this will get simplified when `usng4j` is improved.